### DISCUSSION TITLE
Get Submitted Data on OnRecordAfterCreateRequest Hook

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [pocketbase/pocketbase#2403] [Q&A]
I have two collection with fields
1. Members - name, contactNo
2. payments - member(single relationship field with member), paidDate, amount

While creating members there is an option to make payment or can be created without payment in front end. In case of payment, I need to store it in payments collection. I thought of using OnRecordAfterCreateRequest hook for it so that request data for membership creation will be in following format
```json
{
"name":"John",
"contactNo":"8765468",
"isPaid":true,
"paidDate":"2022-07-02 09:01:34.410Z",
"amount":100.00
}
```
By using after create hook , i need to get raw request data and then in case isPaid is true then i need to create records in payments collection with relation to member created.
How to get raw request data in OnRecordAfterCreateRequest hook?

**Comment by ganigeorgiev** [comment]
> How to get raw request data in OnRecordAfterCreateRequest hook?

This is not documented very well at the moment (it is mentioned only in the godoc), but you can get a copy of the submitted raw request data using the `apis.RequestData(e.HttpContext)` helper, something like:

```go
app.OnRecordAfterCreateRequest("members").Add(func(e *core.RecordCreateEvent) error {
	data := apis.RequestData(e.HttpContext).Data

	// it should be a map with the submitted request body data...
	log.Println(data)

	return nil
})
```
